lib/iris/fileformats/_nc_load_rules/actions.py

```diff
@@ -343,7 +343,42 @@ def action_build_dimension_coordinate(engine, providescoord_fact):
         # Non-conforming lon/lat/projection coords will be classed as
         # dim-coords by cf.py, but 'action_provides_coordinate' will give them
         # a coord-type of 'miscellaneous' : hence, they have no coord-system.
-        coord_system = engine.cube_parts.get("coordinate_system")
+        #
+        # At this point, we need to match any "coordinate_system" entries in
+        # the engine to the coord we are building. There are a couple of cases here:
+        #  1. Simple `grid_mapping = crs` is used, in which case
+        #     we should just apply that mapping to all dim coords.
+        #  2. Extended `grid_mapping = crs: coord1 coord2 crs: coord3 coord4`
+        #     is used in which case we need to match the crs to the coord here.
+
+        # We can have multiple coordinate_system, so now stored as a list (note plural key)
+        coord_systems = engine.cube_parts.get("coordinate_systems")
+        coord_system = None
+
+        if len(coord_systems):
+            # Find which coord system applies to this coordinate.
+            # Parse the grid_mapping attribute to get coord_system -> coordinate mappings
+            attr_grid_mapping = getattr(engine.cf_var, "grid_mapping", None)
+            if attr_grid_mapping:
+                cs_mappings = hh._parse_extened_grid_mapping(attr_grid_mapping)
+
+                # Simple `grid_mapping = "crs"`
+                # Only one coord_system will be present and cs_grid_mapping will
+                # contain no coordinate references (set to None).
+                if len(coord_systems) == 1 and cs_mappings[0][1] is None:
+                    # Only one grid mapping - apply it.
+                    coord_system = list(coord_systems.values())[0]
```

Comment:
```suggestion
                    (coord_system,) = coord_systems.values()
```

Reply:
I knew there was a neater way ... thank you! 💯
6beb20d68